Makefile

```
@@ -94,3 +94,15 @@ release-staging: build
 	docker push $(REGISTRY)/admission-server:$(TAG)
 	docker tag $(REGISTRY)/admission-server:$(TAG) $(REGISTRY)/admission-server:latest
 	docker push $(REGISTRY)/admission-server:latest
+
+# Generate a virtualenv install, which is useful for hacking on the
+# docs since it installs mkdocs and all the right dependencies.
+#
+# On Ubuntu, this requires the python3-venv package.
+virtualenv: .venv
+.venv: requirements.txt
```

```suggestion
.venv:
```

This worked well except one thing:
The first time I ran this, I had an error because I didn't have a depdendency installed.
Then I ran it again, it said nothing to do because the dependent file hadn't changed.

---

What was your error? I can't seem to get `venv` uninstalled from my python :)

---

OK I can reproduce. I added a check to remove the partial venv on error.